SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -29,54 +29,50 @@ using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 template <typename DataT>
 using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
 
-// Calls simd constructor in provided invocation context, which depends on the
-// TestCaseT type. TestCaseT is a struct, that should have call_simd_ctor method
-// that should return constructed object of simd class.
-// This function returns std::vector instance with the output data.
+// Function that verifies obtained test result.
 template <typename DataT, int NumElems, typename TestCaseT>
-auto call_simd(sycl::queue &queue, const shared_vector<DataT> &ref_data) {
-
-  shared_vector<DataT> result{NumElems, shared_allocator<DataT>{queue}};
-
-  queue.submit([&](sycl::handler &cgh) {
-    const auto ref = ref_data.data();
-    auto out = result.data();
+bool verify_test_result(const std::vector<DataT> &ref_data,
+                        const shared_vector<DataT> &result_data,
+                        const std::string &data_type) {
+  bool passed{true};
+
+  for (size_t i = 0; i < result_data.size(); ++i) {
+    if (!are_bitwise_equal(ref_data[i], result_data[i])) {
+      passed = false;
+      log::fail<NumElems>(
+          "Simd by " + TestCaseT::get_description() +
+              " failed, retrieved: " + std::to_string(result_data[i]) +
+              ", expected: " + std::to_string(ref_data[i]) +
+              ", at index: " + std::to_string(i),
+          data_type);
+    }
+  }
 
-    cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
-        [=]() SYCL_ESIMD_KERNEL {
-          sycl::ext::intel::experimental::esimd::simd<DataT, NumElems>
-              result_simd =
-                  TestCaseT::template call_simd_ctor<DataT, NumElems>(ref);
-          result_simd.copy_to(out);
-        });
-  });
-  return result;
+  return passed;
 }
 
 // The main test routine.
 // Using functor class to be able to iterate over the pre-defined data types.
 template <typename DataT, int NumElems, typename TestCaseT> struct test {
-  bool operator()(sycl::queue &queue, const std::string &data_type) {
+  template <typename VerifierT>
```

Comment:
So if you are using the template parameter anyway, why do we need to instantiate the instance?
Or why not just use an abstract base class?

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/614/commits/e44cd0f653ddd3f6c73aa85230716eb71cad054e#diff-a1a30a06ba7da50ce8e13a9af9caec10d385910429792a1de956a7f15cc3d646R68-R71)